src/robot/running/bodyrunner.py

```diff
@@ -610,18 +621,23 @@ def create(cls, limit, variables):
             if count <= 0:
                 raise DataError(f"Invalid WHILE loop limit: Iteration count must be "
                                 f"a positive integer, got '{count}'.")
-            return IterationCountLimit(count)
+            return IterationCountLimit(count, on_limit_message)
         try:
             secs = timestr_to_secs(value)
         except ValueError as err:
             raise DataError(f'Invalid WHILE loop limit: {err.args[0]}')
         else:
-            return DurationLimit(secs)
+            return DurationLimit(secs, on_limit_message)
 
-    def limit_exceeded(self):
-        raise ExecutionFailed(f"WHILE loop was aborted because it did not finish "
-                              f"within the limit of {self}. Use the 'limit' argument "
-                              f"to increase or remove the limit if needed.")
+    def limit_exceeded(self, on_limit_message):
```

Comment:
Why is the message passed here? Couldn't it be accessed like `self.on_limit_message`?